File: chain/java/MasterChef.java
package chain.java;

public class MasterChef implements Chef {
    private Chef nextChef;

    @Override
    public Chef setNextChef(Chef chef) {
        this.nextChef = chef;
        return chef;
    }

    @Override
    public void cook(String dish) {
        System.out.println("MasterChef is overseeing the cooking of: " + dish);
        if (nextChef != null) {
            nextChef.cook(dish);
        } else {
            System.out.println("No chef available to cook: " + dish);
        }
    }
}

File: chain/rust/chain/src/master_chef.rs
use crate::chef::Chef;
use crate::basic_chef::BasicChef;

pub struct MasterChef {
    pub base: BasicChef,
}

impl Chef for MasterChef {
    fn set_next(&mut self, next: Box<dyn Chef>) { self.base.next = Some(next); }
    fn cook(&self, dish: &str) {
        println!("Master Chef performing final garnish on {}!", dish);
        // End of chain; no call to next
    }
}
